src/features/hooks/useDeviceState.js

```diff
@@ -0,0 +1,39 @@
+import React, { useEffect, useState } from "react";
+
+const MIN_MOBILE_WIDTH = 375;
+const MIN_TABLET_WIDTH = 768;
+const MIN_DESKTOP_WIDTH = 1200;
+
+const DEVICE = {
+  MOBILE: "mobile",
+  TABLET: "tablet",
+  DESKTOP: "desktop",
+};
+
+export default function useDeviceState() {
+  const [deviceState, setDeviceState] = useState(getDeviceState());
+
+  useEffect(() => {
+    function handleResize() {
+      window.requestAnimationFrame(() => {
+        setDeviceState(getDeviceState());
+      });
+    }
+
+    window.addEventListener("resize", handleResize);
+
+    return () => {
+      window.removeEventListener("resize", handleResize);
+    };
+  }, []);
+
+  return deviceState;
```

Comment:
오호 밑에 보니까 mobile detect 전용 커스텀 훅을 나눠서 쓰시는것보다는 
이 훅에서 결정해서 리턴값으로 추가해주는 형태는 어때요?
```suggestion
  const [isMobileWidth, setIsMobileWidth] = useState(false);
  
  useEffect(() => {
    setIsMobileWidth(deviceState === "mobile");
  }, [deviceState]);
  
  return { deviceState, isMobile: isMobileWidth }
```

Reply:
[fix(useDeviceState.js): device 너비 판단 hook 병합](https://github.com/alexgoni/6-Sprint-Mission/commit/8c87b91155cbecf94ef68ed9114aacb16899bbcd)

hook이 너무 많아 지는 것이 아닌가 고민이 있었는데
피드백 주신 대로 반영해서 하나 줄일 수 있었습니다!
감사합니다